### ISSUE TITLE
[java] Deprecate UseObjectForClearerAPI

### ISSUE THREAD

**Issue by oowekyala** [pmd/pmd#3741]
The rule [UseObjectForClearerAPI](https://pmd.github.io/latest/pmd_rules_java_design.html#useobjectforclearerapi) reports methods that take more than 3 String parameters. I think the justification is ok, but it's bizarre to single out strings. OTOH, generalizing this to any type of argument just gives [ExcessiveParameterList](https://pmd.github.io/latest/pmd_rules_java_design.html#excessiveparameterlist). Although it's less thoroughly justified, ExcessiveParameterList also recommends using an object to wrap parameters.

Another problem with UseObjectForClearerAPI is that it doesn't care about constructors. Clearly,
```java
callFun("a", "b", "c", "d")
```
is no less readable than
```java
callFun(new FunArguments("a", "b", "c", "d"))
```
So to stay true to its goal of increasing readability, the rule should flag constructors, and also not simply recommend using an object, but also maybe the builder pattern. Since this is burned into the rule name we can't really change that (the rule name doesn't match our [guidelines](https://pmd.github.io/latest/pmd_devdocs_major_rule_guidelines.html)).

----

I think UseObjectForClearerAPI should be deprecated and its description used to enrich the description of ExcessiveParameterList.

**Comment by jtotht** [comment]
I just noticed this in the release notes.

> [I]t's bizarre to single out strings.

I think it _did_ make sense to single out strings – one may be tempted to use strings for everything (especially if the data comes from an untyped source like a properties file or a servlet request), which highly increases the risk of swapping parameters without causing compile-time errors. Consider the following signature:

```java
void banUser(String uuid, String duration, String reason, String notify);

banUser("00000000-0000-0000-0000-000000000000", "bad behavior", "1 day", "true"); // Whoops, duration and reason swapped, but the compiler doesn’t notice this
```

with the following one:

```java
void banUser(java.util.UUID uuid, java.time.Duration duration, String reason, boolean notify);

banUser(new UUID(0, 0), "bad behavior", Duration.ofDays(1), true); // Compilation failure due to the signature mismatch
```

Four parameters are not that excessive that I’d want to configure `ExcessiveParameterList` to report that regardless of the type, but four _strings_ in a row are confusing and should be avoided if possible.
